scaaml/capture/scope/scope_base.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Context manager for the scope."""
+
+from abc import ABC, abstractmethod
+
+
+class AbstractSScope(ABC):
+    """Scope context manager."""
+    def __init__(self, samples: int, offset: int, **_):
+        """Create scope context.
+
+        Args:
+          samples: How samples to capture (length of the capture).
+          offset: Number of samples to wait after trigger event occurred before
+            starting recording data.
+          _: CWScope is expected to be initialized using the capture_info
```

Comment:
Suggestion: make it more generic.
For example: derived classes may require additional parameters to properly configure the scope.

Reply:
It still sounds a bit weird to document specificities of Chipwhisperer scope inside an abstract class. That's what my comment is about.

Reply:
I see, thank you. Then I suggest to remove it as we do not use this parameter anywhere (in the last commit). I added one more commit adding missing argument documentation of PicoScope.